core/src/main/java/io/questdb/cairo/ContiguousVirtualMemory.java

```diff
@@ -0,0 +1,615 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2020 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo;
+
+import io.questdb.griffin.engine.LimitOverflowException;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.*;
+import io.questdb.std.str.AbstractCharSequence;
+import io.questdb.std.str.CharSink;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.Closeable;
+
+/**
+ * 
+ * A version of {@link VirtualMemory} that uses a single contiguous memory region instead of pages. Note that it still has the concept of a page such that the contiguous memory region will grow in page sizes.
+ * 
+ * @author Patrick Mackinlay
+ *
+ */
+public class ContiguousVirtualMemory implements Closeable {
+    static final int STRING_LENGTH_BYTES = 4;
+    private static final Log LOG = LogFactory.getLog(ContiguousVirtualMemory.class);
+    private final ByteSequenceView bsview = new ByteSequenceView();
+    private final CharSequenceView csview = new CharSequenceView();
+    private final CharSequenceView csview2 = new CharSequenceView();
+    private final Long256Impl long256 = new Long256Impl();
+    private final Long256Impl long256B = new Long256Impl();
+    private long pageSize;
+    private final int maxPages;
+    private long baseAddress = 0;
+    private long baseAddressHi = 0;
+    private long appendAddress = 0;
```

Comment:
this is a better implementation than legacy virtual memory. Perhaps we should go ahead and replace virtual memory usages with this class.

Additionally class is not fully tested. There are a number of gaps. Using this impl instead of VirtualMemory will subject code to better testing.

Reply:
I have created a use_contiguous branch, will move everything over to ContiguousVirtualMemory there